x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfig.java

```diff
@@ -135,29 +203,63 @@ public static String documentId(String id) {
         private String source;
         private String dest;
         private List<DataFrameAnalysisConfig> analyses;
+        private Map<String, Object> query = Collections.singletonMap(MatchAllQueryBuilder.NAME, Collections.emptyMap());
 
         public String getId() {
             return id;
         }
 
-        public void setId(String id) {
+        public Builder setId(String id) {
             this.id = ExceptionsHelper.requireNonNull(id, ID);
+            return this;
         }
 
-        public void setSource(String source) {
+        public Builder setSource(String source) {
             this.source = ExceptionsHelper.requireNonNull(source, SOURCE);
+            return this;
         }
 
-        public void setDest(String dest) {
+        public Builder setDest(String dest) {
             this.dest = ExceptionsHelper.requireNonNull(dest, DEST);
+            return this;
         }
 
-        public void setAnalyses(List<DataFrameAnalysisConfig> analyses) {
+        public Builder setAnalyses(List<DataFrameAnalysisConfig> analyses) {
             this.analyses = ExceptionsHelper.requireNonNull(analyses, ANALYSES);
+            return this;
+        }
+
+        public Builder setParsedQuery(QueryBuilder query) {
```

Comment:
do we need this any longer? Can this be simplified?

I only see usage in a unit test, which can be simplified as well

Reply:
Will remove